### PR TITLE
Allow profiler routes

### PR DESCRIPTION
**Pull request by jdlabails**
Options for allow profiler and wdt routes for a specific environment or a particular route.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,8 @@ m6_web_domain_user:
     router_parameter: client   # Parameter in the host requirement
     default_user:     public   # User when the parameter is not found
     users_dir:        %kernel.root_dir%/config/users # Directory containing the user configs
+    firewall:
+        allow_debug_route: true # Allows wdt and profiler routes for the concerned environment
```
Comment (mikaelrandy) [inline README.md]: Do not use "wdt and profiler routes" but "web debug toolbar routes" instead

File: README.md [hunk 2]
```diff
@@ -73,6 +75,8 @@ firewall:
             article: false # Override by routes configured with "defaults: {resource: article}"
         route:
             post_comment: false # Override by route name
+        allow_debug_route: true # Allows wdt and profiler routes fior this user
```
Comment (mikaelrandy) [inline README.md]: s/fior/for/

File: EventListener/UserAccessListener.php [hunk 3]
```diff
@@ -15,15 +16,17 @@
...
     /**
      * Constructor
      *
      * @param TokenStorageInterface $tokenStorage
      */
-    public function __construct(TokenStorageInterface $tokenStorage)
+    public function __construct(TokenStorageInterface $tokenStorage, $allowDebugRoute)
     {
         $this->tokenStorage = $tokenStorage;
+        $this->allowDebugRoute = $allowDebugRoute;
```
Comment (mikaelrandy) [inline EventListener/UserAccessListener.php]: Why do you need this ?

Comment (jdlabails) [inline EventListener/UserAccessListener.php]: to get environment option

File: README.md [hunk 4]
```diff
@@ -54,6 +54,8 @@ m6_web_domain_user:
     router_parameter: client   # Parameter in the host requirement
     default_user:     public   # User when the parameter is not found
     users_dir:        %kernel.root_dir%/config/users # Directory containing the user configs
+    firewall:
+        allow_debug_route: true # Allows web debug toolbar routes for the concerned environment
```
Comment (valentin-claras) [inline README.md]: Imho, it's better to use the default value (false) in examples.

File: README.md [hunk 5]
```diff
@@ -73,6 +75,8 @@ firewall:
             article: false # Override by routes configured with "defaults: {resource: article}"
         route:
             post_comment: false # Override by route name
+        allow_debug_route: true # Allows wdt and profiler routes for this user
```
Comment (valentin-claras) [inline README.md]: That here, default is false.

File: README.md [hunk 6]
```diff
@@ -107,7 +111,7 @@ $ ./vendor/bin/atoum
 
 ## Credits
 
-Developped by the [Cytron Team](http://cytron.fr/) of [M6 Web](http://tech.m6web.fr/).  
+Developped by the [Cytron Team](http://cytron.fr/) of [M6 Web](http://tech.m6web.fr/).
```
Comment (valentin-claras) [inline README.md]: Thoses links are seriously outdated :lollipop:

**Comment by valentin-claras** [comment]
It's missing a new test to check 'allow_debug_route' for a specific user and another one about global 'allow_debug_route'.

File: README.md (hunk 4 above)
Comment (mikaelrandy) [inline README.md]: :+1:

**Comment by jdlabails** [comment]
@mikaelrandy @valentin-claras readme review and test for UserAccessListener added

File: Tests/Fixtures/users/user2.yml [hunk 7]
```diff
@@ -6,3 +6,6 @@ entities:
         myflag: 3
 parameters:
     param_name: param_value
+firewall:
+    allow:
+        allow_debug_route: true
```
Comment (mikaelrandy) [inline Tests/Fixtures/users/user2.yml]: missing end of file

Comment (jdlabails) [inline Tests/Fixtures/users/user2.yml]: done

**Comment by mikaelrandy** [comment]
:+1:

**Comment by valentin-claras** [comment]
:+1:

**Pull request merged**